### ISSUE TITLE
Beam search validation

### ISSUE THREAD

**Issue by yellowjs0304** [airsplay/R2R-EnvDrop#26]
Hi, Is this any way to test your model as beam search?

I saw the code in train.py and agent.py ("beam_valid", "beam_search_test")

Could you please share any way to use it?

**Comment by airsplay** [comment]
Thanks. Please check [this issue](https://github.com/airsplay/R2R-EnvDrop/issues/11) for a reference.

**Comment by yellowjs0304** [comment]
thank u for reply. I'm sorry it's too late..
I already saw that and It seems you already updated the beam search code in github with named "_dijkstra"
I have confusing with the code "_dijkstra" function. 
why is this named _dijkstra? i thought the beam search is only rely on agent's logits. but in your code it seems it's related with the dijkstra path.

It seems i haven't quite understood your system.

**Comment by airsplay** [comment]
> why is this named _dijkstra? i thought the beam search is only rely on agent's logits. but in your code it seems it's related with the dijkstra path.

This algorithm still follows the beam-search approach mentioned in the [Speaker-Follower paper](https://arxiv.org/abs/1806.02724). I modified it to a greedy approach ([Dijkstra algorithm](https://en.wikipedia.org/wiki/Dijkstra%27s_algorithm)) that only expands the state with the lowest score at each time step. It would empirically have a better estimation of the best path than the beam-search approach without losing speed.

**Comment by yellowjs0304** [comment]
Okay I understood. If so, is it affected with the agent(listener) model's tokenizer?
Because I tried with my own models which has bert embedding layer, and it returns really low success rate with beam search...
In greedy search it doesn't that so low.
Could you please give me some opinion about this?
